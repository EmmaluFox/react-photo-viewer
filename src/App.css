.App {
    max-width: 1200px;
    margin: auto;
    padding: 0 8px;
}