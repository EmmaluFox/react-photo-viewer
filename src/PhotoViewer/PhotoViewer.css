.selected-photo {
    width: 100%;
}