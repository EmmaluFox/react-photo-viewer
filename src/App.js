import React, {useState} from 'react';
import './App.css';
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import {PhotoSelector} from "./PhotoSelector/PhotoSelector";
import {ImageUrls} from "./data";

function App() {
    const [selectedUrl, setSelectedUrl] = useState(ImageUrls[0]);
    
    return (
        <div className="App">
            <h1>React Photo Viewer</h1>
            <PhotoViewer src={selectedUrl}/>
            <PhotoSelector selectedUrl={selectedUrl} setSelectedUrl={setSelectedUrl}/>
        </div>
    );
}

export default App;
