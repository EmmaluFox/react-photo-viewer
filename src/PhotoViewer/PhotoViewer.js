import React from "react";
import PropTypes from "prop-types";
import "./PhotoViewer.css";

function PhotoViewer(props) {
    return (
        <section>
            <img className="selected-photo" src={props.src} alt=""/>
        </section>
    );
}

PhotoViewer.propTypes = {
    src: PropTypes.string.isRequired,
};

export {PhotoViewer};