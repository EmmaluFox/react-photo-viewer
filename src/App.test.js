import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import {ImageUrls} from "./data";
import {PhotoViewer} from "./PhotoViewer/PhotoViewer";
import TestRenderer from 'react-test-renderer';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('count number of urls', () => {
  //arrange
  const imageUrls = ImageUrls;
  const arrayLength = imageUrls.length;
  //act
  
  //assert
  expect(arrayLength).toBe(44);
  
});
describe("PhotoViewer", () => {
  it("Should not change", async () => {
    const tree = TestRenderer.create(<PhotoViewer src={ImageUrls[0]}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

});
