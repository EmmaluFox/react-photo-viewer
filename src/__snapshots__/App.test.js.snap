// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoViewer Should not change 1`] = `
<section>
  <img
    alt=""
    className="selected-photo"
    src="https://i.picsum.photos/id/600/1600/900.jpg"
  />
</section>
`;
